### PR TITLE
CNAPP-10978-Change-all-TF-SDK-API-calls-to-use-the-new-retry-function

### PR DESCRIPTION
**Pull request by idorosCP**

### CODE REVIEW

File: services/cloudaccounts/aws/aws.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (service *Service) ForceDelete(id string) (*http.Response, error) {
 	var err error
 
 	for i := 1; i <= 3; i++ {
-		resp, err = service.Client.NewRequestDo("DELETE", relativeURL, nil, nil, nil)
+		resp, err = service.Client.NewRequestDoRetry("DELETE", relativeURL, nil, nil, nil, nil)
```
Comment (dolevh-d9) [inline services/cloudaccounts/aws/aws.go:191]: There is already a retry mechanism here, please delete it

File: services/integrations/integration.go [hunk 2]
```diff
@@ -146,7 +146,7 @@ func (service *Service) Delete(id string) (*http.Response, error) {
 	var err error
 
 	for i := 1; i <= 3; i++ {
-		resp, err = service.Client.NewRequestDo("DELETE", relativeURL, nil, nil, nil)
+		resp, err = service.Client.NewRequestDoRetry("DELETE", relativeURL, nil, nil, nil, nil)
```
Comment (dolevh-d9) [inline services/integrations/integration.go:149]: There is already a retry mechanism here, please delete it

File: dome9/client/client.go [hunk 3]
```diff
@@ -48,7 +48,11 @@ func (client *Client) NewRequestDoRetryWithOptions(method, url string, options,
 			// If the request was successful, return the response
 			return resp, nil
 		}
-
+		// If status code is 429 (API throttling), set the retrySleepBetweenSecs to 10 and maxRetries to fit 5 minutes of retries
+		if resp.StatusCode == http.StatusTooManyRequests {
```
[the remark this answers is not in the document]
Comment (dolevh-d9) [reply dome9/client/client.go:52]: if what isn't 0?

**Review by dolevh-d9 — CHANGES_REQUESTED** [review]
Note the notes

File: dome9/client/client.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (idorosCP) [reply dome9/client/client.go:52]: 'i' from the for loop, meaning for example if we received 429 on the third 'NewRequestDo' attempt, retrySleepBetweenSecs and maxRetries wont align to exactly 5 minutes. Is this a edge case I should handle?

[the remark this answers is not in the document]
Comment (dolevh-d9) [reply dome9/client/client.go:52]: No

**Review by dolevh-d9 — APPROVED** [review]
Great
Please check with the provider

**Pull request merged**